src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -543,7 +543,7 @@ public void reset() throws IOException {
 
     private void implReset() throws IOException {
         getBufIfOpen(); // Cause exception if closed
-        if (markpos < 0)
+        if (markpos == -1)
```

Comment:
Just looking at this one again and it might be better to not change implReset. If a broken subclass were to set markpos to less than -1 then the reset() method would throw and avoid a more confusing IIOOBE or other exception.

Reply:
This is true, but if a broken subclass sets markpos to less than -1 then lots of code lines will work incorrectly -- including `transferTo` which was the starting point of your change proposal. So do you really only want undo the change *here*, or is it better drop this PR and even use `< 0` in `transferTo`, as I originally proposed?

Reply:
> This is true, but if a broken subclass sets markpos to less than -1 then lots of code lines will work incorrectly -- including `transferTo` which was the starting point of your change proposal. So do you really only want undo the change _here_, or is it better drop this PR and even use `< 0` in `transferTo`, as I originally proposed?

This looks to be the only that would corrupt the current position (pos).

Reply:
@AlanBateman Checking `< 0` just here, as per your proposal. 👍 

@bplb Kindly requesting `/sponsor`. 😃

Reply:
> Checking `< 0` just here, as per your proposal. 👍

In the most recent webrev, in `implReset` it still has
```if (markpos == -1)```
Was that supposed to be changed back to `< 0`?

Reply:
I don't see any problems here, but I question to a degree the value of the change which is not really fixing any observed broken behavior but cosmetically restricting `markpos` to the specified range `[-1, pos]` instead of `[Integer.MIN_VALUE, pos]`.

Reply:
This PR originates from @AlanBateman who asked me to change my use of `== 1` to `< 0`. Per his opinion, using `< 0` is allegedly wrong, as it violates the spec of `markpos`. As it either it is wrong *everywhere* or *nowhere*, he agreed that this should be fixed. So guys, please discuss with each other what way you finally want it to be like. Thanks.

Reply:
> This PR originates from @AlanBateman who asked me to change my use of `== 1` to `< 0`. Per his opinion, using `< 0` is allegedly wrong, as it violates the spec of `markpos`. As it either it is wrong _everywhere_ or _nowhere_, he agreed that this should be fixed. So guys, please discuss with each other what way you finally want it to be like. Thanks.

I am aware of the context and agree that `-1 <= markpos <= pos` should be `true`, but am simply pointing out that it does not really fix an observable problem and no change is without risk.